src/mlpack/bindings/R/print_R.cpp

```diff
@@ -55,60 +55,147 @@ void PrintR(const util::ProgramDoc& programInfo,
 
   for (ParamIter it = parameters.begin(); it != parameters.end(); ++it)
   {
-    const util::ParamData& d = it->second;
+    util::ParamData& d = it->second;
     if (d.input && !d.required &&
         d.name != "help" && d.name != "info" &&
         d.name != "version")
       inputOptions.push_back(it->first);
   }
 
+  // Print the documentation.
+  // Print programName as @title.
+  cout << "#' @title ";
+  cout << util::HyphenateString(programInfo.programName, "#'   ") << endl;
+  cout << "#'" << endl;
+
+  // Next print the short description as @description.
+  cout << "#' @description" << endl;
+  cout << "#' ";
+  cout << util::HyphenateString(programInfo.shortDocumentation, "#' ") << endl;
+
+  // Next, print information on the input options.
+  cout << "#'" << endl;
+
+  for (size_t i = 0; i < inputOptions.size(); ++i)
+  {
+    const string& opt = inputOptions[i];
+    util::ParamData& d = parameters.at(opt);
+
+    cout << "#' @param ";
+    bool out = false;
+    IO::GetSingleton().functionMap[d.tname]["PrintDoc"](d, NULL, (void*) &out);
+
+    cout << endl;
+  }
+  cout << "#'" << endl;
+
+  // Next, print information on the output options.
+  cout << "#' @return A list with several components:" << endl;
+
+  for (size_t i = 0; i < outputOptions.size(); ++i)
+  {
+    const string& opt = outputOptions[i];
+    util::ParamData& d = parameters.at(opt);
+
+    cout << "#' \\item{";
+
+    bool out = true;
+    IO::GetSingleton().functionMap[d.tname]["PrintDoc"](d, NULL, (void*) &out);
+
+    cout << endl;
+  }
+  cout << "#'" << endl;
+
+  // Next print the long description as @details.
+  cout << "#' @details" << endl;
+  cout << "#' ";
+  cout << util::HyphenateString(programInfo.documentation(), "#' ") << endl;
+  cout << "#' @author" << endl;
+  cout << "#' MLPACK Developers" << endl;
```

Comment:
```suggestion
  cout << "#' mlpack developers" << endl;
```
(We should use `mlpack` not `MLPACK` to match the "branding" of the rest of the project.  I know, everybody always capitalizes it differently when they talk about it, but at least in the codebase, website, and (recent) papers we are consistent with `mlpack`.  This is the problem with choosing a name that's half an acronym and half a word... :smile:)